libManager/libManager.go

```diff
@@ -0,0 +1,133 @@
+package libManager
+
+import (
+	"path/filepath"
+
+	"github.com/future-architect/vuls/models"
+	"github.com/future-architect/vuls/util"
+	godeptypes "github.com/knqyf263/go-dep-parser/pkg/types"
+	"github.com/knqyf263/go-version"
+	"github.com/knqyf263/trivy/pkg/db"
+	"github.com/knqyf263/trivy/pkg/log"
+	"github.com/knqyf263/trivy/pkg/scanner/library"
+	"github.com/knqyf263/trivy/pkg/vulnsrc/vulnerability"
+
+	"golang.org/x/xerrors"
+)
+
+func init() {
+	log.InitLogger(false)
+	if err := db.Init(); err != nil {
+		util.Log.Debugf("library db cant initialized")
+	}
+}
+
+func FillLibrary(r *models.ScanResult) (totalCnt int, err error) {
+
+	for _, lib := range r.LibraryScanners {
+		vinfos, err := scan(lib.Path, lib.Libs)
+		if err != nil {
+			return 0, err
+		}
+		for _, vinfo := range vinfos {
+			r.ScannedCves[vinfo.CveID] = vinfo
+		}
+		totalCnt += len(vinfos)
+	}
+
+	return totalCnt, nil
+}
+
+func scan(path string, pkgs []godeptypes.Library) ([]models.VulnInfo, error) {
+	scanner := library.NewScanner(filepath.Base(string(path)))
+	if scanner == nil {
+		return nil, xerrors.New("unknown file type")
+	}
+
+	err := scanner.UpdateDB()
+	if err != nil {
+		return nil, xerrors.Errorf("failed to update %s advisories: %w", scanner.Type(), err)
+	}
+
+	var vulnerabilities []models.VulnInfo
+	for _, pkg := range pkgs {
+		v, err := version.NewVersion(pkg.Version)
+		if err != nil {
+			util.Log.Debugf("new version cant detected %s@%s", pkg.Name, pkg.Version)
+			continue
+		}
+
+		tvulns, err := scanner.Detect(pkg.Name, v)
+		if err != nil {
+			return nil, xerrors.Errorf("failed to detect %s vulnerabilities: %w", scanner.Type(), err)
+		}
+
+		vulns := convertFanalToVuln(tvulns)
+		vulnerabilities = append(vulnerabilities, vulns...)
+	}
+
+	return vulnerabilities, nil
+}
+
+func convertFanalToVuln(tvulns []vulnerability.DetectedVulnerability) (vulns []models.VulnInfo) {
+	for _, tvuln := range tvulns {
+		vinfo, _ := getVulnDetail(tvuln)
+		vulns = append(vulns, vinfo)
+	}
+	return vulns
+}
+
+func getVulnDetail(tvuln vulnerability.DetectedVulnerability) (vinfo models.VulnInfo, err error) {
+	details, err := vulnerability.Get(tvuln.VulnerabilityID)
+	if err != nil {
+		return vinfo, err
+	} else if len(details) == 0 {
+		return vinfo, xerrors.Errorf("Unknown vulnID : %s", tvuln.VulnerabilityID)
+	}
+	vinfo.CveID = tvuln.VulnerabilityID
+	vinfo.CveContents = getCveContents(details)
+	if tvuln.FixedVersion != "" {
+		vinfo.PackageFixedIns = []models.PackageFixedIn{
+			{
+				Name:    tvuln.PkgName,
+				FixedIn: tvuln.FixedVersion,
+			},
+		}
+	}
+	return vinfo, nil
+}
+
+func getCveContents(details map[string]vulnerability.Vulnerability) (contents map[models.CveContentType]models.CveContent) {
+	contents = map[models.CveContentType]models.CveContent{}
+	for source, detail := range details {
+		refs := []models.Reference{}
+		for _, refUrl := range detail.References {
```

Comment:
__[Golint]__ range var refUrl should be refURL ([link](https://sider.review/gh/repos/54808920/pulls/829#issue-4176676))

<sub>posted by Sider</sub>